astropy/io/tests/test_registry.py

```diff
@@ -274,6 +277,13 @@ def test_read_invalid_return():
     assert exc.value.args[0] == "reader should return a TestData instance"
 
 
+def test_non_existing_unknown_ext():
+    """Raise the correct error when attempting to read a non-existing
+    file with an unknown extension."""
+    with pytest.raises(IOError):
+        data = Table.read('non-existing-file-with-unknown.ext')
```

Comment:
Not really worth changing, but some linters will balk at this statement that sets `data` but never uses it.  In cases like this you just do `Table.read(...)` and expect the exception.